logic/transaction_pool.js

```diff
@@ -885,53 +895,56 @@ __private.processVerifyTransaction = function(transaction, broadcast, cb, tx) {
  * @returns {SetImmediate} error
  */
 __private.applyUnconfirmedList = function(transactions, cb, tx) {
-	async.eachSeries(
-		transactions,
-		(transaction, eachSeriesCb) => {
-			if (!transaction) {
-				return setImmediate(eachSeriesCb);
-			}
-			library.balancesSequence.add(balancesSequenceCb => {
-				__private.processVerifyTransaction(
-					transaction,
-					false,
-					(err, sender) => {
-						if (err) {
-							library.logger.error(
-								`Failed to process / verify unconfirmed transaction: ${
-									transaction.id
-								}`,
-								err
+	library.balancesSequence.add(
+		balancesSequenceCb => {
+			async.eachSeries(
+				transactions,
+				(transaction, eachSeriesCb) => {
+					if (!transaction) {
+						return setImmediate(eachSeriesCb);
+					}
+					__private.processVerifyTransaction(
+						transaction,
+						false,
+						(processVerifyErr, sender) => {
+							if (processVerifyErr) {
+								library.logger.error(
+									`Failed to process / verify unconfirmed transaction: ${
+										transaction.id
+									}`,
+									processVerifyErr
+								);
+								self.removeQueuedTransaction(transaction.id);
+								return setImmediate(eachSeriesCb);
+							}
+							modules.transactions.applyUnconfirmed(
+								transaction,
+								sender,
+								applyUnconfirmErr => {
+									if (applyUnconfirmErr) {
+										library.logger.error(
+											`Failed to apply unconfirmed transaction: ${
+												transaction.id
+											}`,
+											applyUnconfirmErr
+										);
+										self.removeQueuedTransaction(transaction.id);
+									} else {
+										// Transaction successfully applied to unconfirmed states, move it to unconfirmed list
+										self.addUnconfirmedTransaction(transaction);
+									}
+									return setImmediate(eachSeriesCb);
+								},
+								tx
 							);
-							self.removeQueuedTransaction(transaction.id);
-							return setImmediate(balancesSequenceCb);
-						}
-						modules.transactions.applyUnconfirmed(
-							transaction,
-							sender,
-							err => {
-								if (err) {
-									library.logger.error(
-										`Failed to apply unconfirmed transaction: ${
-											transaction.id
-										}`,
-										err
-									);
-									self.removeQueuedTransaction(transaction.id);
-								} else {
-									// Transaction successfully applied to unconfirmed states, move it to unconfirmed list
-									self.addUnconfirmedTransaction(transaction);
-								}
-								return setImmediate(balancesSequenceCb);
-							},
-							tx
-						);
-					},
-					tx
-				);
-			}, eachSeriesCb);
+						},
+						tx
+					);
+				},
+				balancesSequenceCb
+			);
 		},
-		cb
+		() => setImmediate(cb)
```

Comment:
Why we are not sending the `seriesCallBackErr` here?

Reply:
If we return cb directly from here or pass seriesCbErr, it starts breaking the tests. Due to  `addTransactionsAndForge` is written in tests. Again, didn't want to fix irrelevant issues in this PR.